controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    # Durum yayıncısı (Robotun nerede olduğunu söyler)
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    # Hareket kontrolcüsü (Bizim emirlerimizi uygular)
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_trajectory_controller:
  ros__parameters:
    joints:
      - base_to_rotary
      - rotary_to_lower
      - lower_to_upper
      - upper_to_support
      - support_to_wrist
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    open_loop_control: true
    allow_partial_joints_goal: true